baybe/utils/scaling.py

```diff
@@ -2,21 +2,57 @@
 
 from __future__ import annotations
 
-from typing import Protocol
-
-import pandas as pd
-
-
-class ParameterScalerProtocol(Protocol):
-    """Type protocol specifying the interface parameter scalers need to implement.
-
-    The protocol is compatible with sklearn scalers such as
-    :class:`sklearn.preprocessing.MinMaxScaler` or
-    :class:`sklearn.preprocessing.MaxAbsScaler`.
-    """
-
-    def fit(self, df: pd.DataFrame, /) -> None:
-        """Fit the scaler to a given dataframe containing parameter configurations."""
-
-    def transform(self, df: pd.DataFrame, /) -> pd.DataFrame:
-        """Transform a parameter dataframe using the fitted scaling logic."""
+import itertools
+from typing import TYPE_CHECKING
+
+from attrs import define, field
+from attrs.validators import deep_iterable, deep_mapping, instance_of
+
+if TYPE_CHECKING:
+    from botorch.models.transforms.input import InputTransform
+    from torch import Tensor
+
+
+@define
+class ColumnTransformer:
+    """Class for applying transforms to individual columns of tensors."""
```

Comment:
Isn't the whole idea of the transformer to apply transforms to a collection of columns? This reads exactly the other way.

Reply:
At the lowest level, each column is still transformed independently of the others (which made me write it this way). But I agree that this could be made more precise. Have changed it to `Class for applying separate transforms to different column groups of tensors.`